picodom.d.ts

```diff
@@ -1,23 +1,38 @@
-export interface VNode<Data> {
+export as namespace picodom
+
+export interface VNode<Props> {
   type: string
   props?: Props
-  children: VNode<Props> | string
+  children: Array<VNode<{}> | string>
 }
 
-export interface Component<Props> {
-  (props?: Props, ...children: VNode<Props> | string): VNode<Props>
+export interface Component<Props = {}> {
+  (props: Props, children: Array<VNode<{}> | string>): VNode<Props>
 }
 
 export function h<Props>(
   type: Component<Props> | string,
   props?: Props,
-  ...children: VNode<Props> | string
+  ...children: Array<VNode<{}> | string | number | null>
+): VNode<Props>
+
+export function h<Props>(
+  tag: Component<Props> | string,
+  data?: Props,
+  children?: Array<VNode<{}> | string | number | null>
 ): VNode<Props>
```

Comment:
@andrewiggins Why is this repeated?

```js
export function h<Props>(
  type: Component<Props> | string,
  props?: Props,
  ...children: Array<VNode<{}> | string | number | null>
): VNode<Props>

export function h<Props>(
  tag: Component<Props> | string,
  data?: Props,
  children?: Array<VNode<{}> | string | number | null>
): VNode<Props>
```

Reply:
Overloading. One accepts spread arguments `...children` and the other accepts an array.

Reply:
Right, would it be possible to express that in the same export declaration?

Reply:
Type/tag and data/props could be consistently named in those declarations though. (and throughout the file.)

Reply:
Alternatively, if that is not possible, could we remove the declaration that takes spread arguments without TypeScript complaining and leave it in the source just as a detail?

Reply:
I don't think you can put those in a single declaration, no - they're different call signatures because they have different number of arguments. (Typescript validates overloads for compatibility though, so the duplication is not a liability.)

Reply:
I don't think you should remove the spread declaration, since that's the one that Typescript output will actually use.

Reply:
Gotcha! 👍 

Now, people using `h()` should always express children using an array. The spread arguments syntax is supported only because JSX needs it, but this may change as JSX evolves / improves.

Reply:
@mindplay-dk I thought the order was: TypeScript -> JSX -> JavaScript